### PR TITLE
use conda build api for actual build

### PR DESCRIPTION
**Pull request by cdeepali**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [x] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

Fixes https://github.com/open-ce/open-ce/issues/53.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: tests/build_feedstock_test.py [hunk 1]
```diff
@@ -53,14 +54,12 @@ def test_build_feedstock_failure(mocker, capsys):
         return_value=False
     )
     mocker.patch(
-        'os.system',
-        side_effect=(lambda x: helpers.validate_cli(x, expect=["conda-build"], retval=1)) # Retval is 1 to simulate a failure.
+        'build_feedstock.build_feedstock',
```
Comment (cdeepali) [inline tests/build_feedstock_test.py:57]: I am checking if this can be replaced with any other check to detect conda_build.api.build failure.

Comment (cdeepali) [reply tests/build_feedstock_test.py]: Updated to mock `conda_buld.api.build` to throw a `ValueError` to simulate a failure.   Thanks @bethune-bryant  for suggesting this approach.

File: open-ce/build_feedstock.py [hunk 2]
```diff
@@ -157,32 +160,31 @@ def build_feedstock(args_string=None):
...
-            if result != 0:
-                break
-
-            print(conda_build_args)
-            result = os.system(conda_build_args)
-            if result != 0:
-                print("Failure building recipe: " + (recipe['name'] if 'name' in recipe else os.getcwd))
-                result = 1
-                break
+
```
Comment (bnemanich) [inline open-ce/build_feedstock.py:163]: This needs to be merged with the changes added in PR #90 . There should now be a loop going over the different variants.

Comment (cdeepali) [reply open-ce/build_feedstock.py:163]: yes have added it - https://github.com/open-ce/open-ce/blob/1a8f249a57288e015d56c345c63cbbc430a46d33/open-ce/build_feedstock.py#L180

**Review by bnemanich — APPROVED** [review]
Looks good!

**Review by bethune-bryant — APPROVED** [review]
This looks great!
Thanks @cdeepali !

**Comment by cdeepali** [comment]
Thanks @bnemanich, @bethune-bryant. Merging it now.

**Pull request merged**